src/index/src/bloom_filter/reader.rs

```diff
@@ -33,6 +34,45 @@ const BLOOM_META_LEN_SIZE: u64 = 4;
 /// Default prefetch size of bloom filter meta.
 pub const DEFAULT_PREFETCH_SIZE: u64 = 8192; // 8KiB
 
+/// Safely converts bytes to Vec<u64> using bytemuck for optimal performance.
+/// Faster than chunking and converting each piece individually.
+fn bytes_to_u64_vec(bytes: &Bytes) -> Vec<u64> {
```

Comment:
Can we write some benchmarks (aligned and unaligned) for this against the old implementation?